client/packages/core/src/instaql.js

```diff
@@ -258,21 +260,34 @@ function extendObjects(makeVar, store, { etype, level, form }, objects) {
   return Object.entries(objects).map(([eid, parent]) => {
     const childResults = children.map((label) => {
       try {
-        const [nextEtype, nextLevel, join] = makeJoin(
+        const [nextEtype, nextLevel, join, attr, isForward] = makeJoin(
           makeVar,
           store,
           etype,
           level,
           label,
           eid,
         );
-        const child = queryOne(store, {
+        const childrenArray = queryOne(store, {
           etype: nextEtype,
           level: nextLevel,
           form: form[label],
           join,
         });
-        return { [label]: child };
+
+        let children = childrenArray;
+        if (store.schema) {
```

Comment:
cc @markyfyi instead of using schema as the flag, I would use cardinalityInference directly. 

Would also extract this bit into a helpful function